docs/types.rst

```diff
@@ -857,13 +859,31 @@ or create a new memory array and copy every element.
         }
     }
 
-.. index:: ! array;literals, !inline;arrays
+.. index:: ! array;literals, ! inline;arrays
 
-Array Literals / Inline Arrays
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Array Literals
+^^^^^^^^^^^^^^
 
-Array literals are arrays that are written as an expression and are not
-assigned to a variable right away.
+Array literals (also called inline arrays) are arrays that are written as an
+expression and are not assigned to a variable right away.
+
+::
+
+  pragma solidity >=0.4.16 <0.6.0;
+
+  contract C {
+      function f() public pure {
+          g([uint(1), 2, 3]);
+      }
+      function g(uint[3] _data) public pure {
+          // ...
+      }
+  }
+
+The type of an array literal is a memory array of fixed size whose base
+type is the common type of the given elements. For example, the type of ``[1, 2, 3]`` is
+``uint8[3] memory``, because the type of each of these constants is ``uint8``,
+it is necessary to convert the first element to ``uint``.
```

Comment:
It is ONLY necessary to convert it if you want the result to be a `uint[3] memory`.

Reply:
Updated, hopefully I understood you correctly.